src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -1543,751 +1493,748 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
+      ! End preamble
+      !-----------------------------------------------------------------
+      ! Begin code
 
-      ! Determine the number of barotropic subcycles based on the ratio of time steps
-      call mpas_pool_get_config(domain % configs, 'config_time_integrator', config_time_integrator)
-      call mpas_pool_get_config(domain % configs, 'config_btr_dt', config_btr_dt)
-      call mpas_pool_get_config(domain % configs, 'config_dt', config_dt)
+      !*** Determine the number of barotropic subcycles based on the
+      !*** ratio of time steps
 
-      nowTime = mpas_get_clock_time(domain % clock, MPAS_NOW, ierr)
-      call mpas_set_timeInterval( zeroInterval, S=0 )
+      nowTime = mpas_get_clock_time(domain%clock, MPAS_NOW, ierr)
 
+      call mpas_set_timeInterval( zeroInterval, S=0 )
       call mpas_set_timeInterval( fullTimeStep , timeString=config_dt )
-      call mpas_set_timeInterval( barotropicTimeStep, timeString=config_btr_dt )
+      call mpas_set_timeInterval( barotropicTimeStep, &
+                                             timeString=config_btr_dt )
 
-      ! transfer to I8 for division step
+      !*** transfer to I8 for division step
       nBtrSubcyclesI8 = nBtrSubcycles
-      call mpas_interval_division( nowTime, fullTimeStep, barotropicTimeStep, nBtrSubcyclesI8, remainder )
+      call mpas_interval_division( nowTime, fullTimeStep, &
+                       barotropicTimeStep, nBtrSubcyclesI8, remainder )
       nBtrSubcycles = nBtrSubcyclesI8
 
       if ( remainder > zeroInterval ) then
          nBtrSubcycles = nBtrSubcycles + 1
       end if
 
-      if (trim(config_time_integrator) == 'split_explicit') then
+      !*** Set mask for using velocity correction
+      if (config_vel_correction) then
+         useVelocityCorrection = 1.0_RKIND
+      else
+         useVelocityCorrection = 0.0_RKIND
+      endif
+
+      !*** Compute some halo needs to reduce the number
+      !*** of halo updates during subcycling
+      neededHalos = 1 + config_n_btr_cor_iter
+      haloDecrement = mod(config_num_halos, neededHalos)
+
+      !*** Determine the time integration type and set associated masks
+
+      select case (trim(config_time_integrator))
+
+      case ('unsplit_explicit')
+         unsplit   = .true.
+         splitFact = 0.0_RKIND
+
+      case ('split_explicit')
+         unsplit   = .false.
+         splitFact = 1.0_RKIND
          call mpas_log_write( '*******************************************************************************')
          call mpas_log_write( 'The split explicit time integration is configured to use: $i barotropic subcycles', &
             intArgs=(/ nBtrSubcycles /) )
          call mpas_log_write( '*******************************************************************************')
-      end if
 
-      if ( .not. config_do_restart ) then
-         ! Initialize z-level mesh variables from h, read in from input file.
+      case default
```

Comment:
This causes a FAIL in the RK4 test cases.  It appears that `ocn_time_integration_split_init` is called no matter what the time integrator is ([see line 265 in `mpas_ocn_forward_mode.F`](https://github.com/philipwjones/MPAS-Model/blob/ocean/sedriver/src/core_ocean/mode_forward/mpas_ocn_forward_mode.F#L261-L267))
```
      call ocn_init_routines_vert_coord(domain)

      call ocn_init_routines_compute_max_level(domain)

      call ocn_time_integration_split_init(domain)

      call mpas_log_write(' Vertical coordinate movement is: ' // trim(config_vert_coord_movement))
```

@philipwjones or @mark-petersen Do either of you know if there is a reason why the split-explicit init routine is called no matter the value of `config_time_integrator`?  Or is it OK for me to replace the call with
```
      call mpas_pool_get_config(domain % configs, 'config_time_integrator', config_time_integrator)                
                                                                                                                   
      select case (trim(config_time_integrator))                                                                   
                                                                                                                   
      case ('semi_implicit')                                                                                       
        call ocn_time_integration_si_init(domain)                                                                  
                                                                                                                   
      case ('split_explicit','unsplit_explicit')                                                                   
        call ocn_time_integration_split_init(domain)                                                               
                                                                                                                   
      case ('RK4')                                                                                                 
        ! No init routine needs to be run for RK4                                                                  
                                                                                                                   
      end select
```

Reply:
I just pushed a commit to wrap the calls to time integration initialization routines in a `select case`.